modules/moderation.py

```diff
@@ -45,17 +62,36 @@ def validate_token(token: str) -> bool:
         user_id = int(base64.b64decode(user_id + "==", validate=True))
     except (ValueError, binascii.Error):
         return False
-    else:
-        return True
+    return True
 
 
 class GithubError(commands.CommandError):
     pass
 
 
+class ModerationRespostView(discord.ui.View):
+    def __init__(self, *, timeout: float | None = 180, target_id: int, target_reason: str) -> None:
+        super().__init__(timeout=timeout)
+        self.target: discord.Object = discord.Object(id=target_id, type=discord.Member)
+        self.target_reason: str = target_reason
+
+    @discord.ui.button(label="Ban", emoji="\U0001f528")
+    async def ban_button(self, interaction: Interaction, button: discord.ui.Button[Self]) -> None:
+        assert interaction.guild
+        await interaction.response.defer(ephemeral=False)
+
+        reason = f"Banned due to grievances in discord.py: {self.target_reason!r}"
+        await interaction.guild.ban(
+            self.target,
+            reason=shorten(reason, width=128, placeholder="..."),
+        )
+        await interaction.followup.send("Banned.")
+
+
 class Moderation(commands.Cog):
     def __init__(self, bot: core.Bot, /) -> None:
         self.bot = bot
+        self.dpy_mod_cache: dict[int, discord.User | discord.Member] = {}
```

Comment:
This will never clear, leading to ever-increasing memory usage. perhaps an LRU?

Reply:
Realistically this will never be more than ~3 members big. I think it's safe as a plain dictionary. We could make it a weakref however, and let the GC handle it.

Reply:
Resolving as discussed